sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java

```diff
@@ -36,10 +37,17 @@ public enum HttpLogDetailLevel {
     /**
      * Logs everything in HEADERS and BODY.
      */
-    BODY_AND_HEADERS;
+    BODY_AND_HEADERS,
+
+    /**
+     * Logs everything in BASIC, plus allowed request and response headers.
+     */
+    ALLOWED_HEADERS;
```

Comment:
Is this the way we want to handle this? Would it be better as an option in `HttpLogOptions` such as `includeRedactedHeadersInLogs`. As it is right now, this wouldn't work when using `BODY_AND_HEADERS` and I could see a case where it'd be used there as well.

Additionally, do we want to have a similar handling for query parameters?

Reply:
Yeah, I think we should move this to HttpLogOptions. Since we already have HEADERS and BODY_AND_HEADERS and this option to log/skip redacted headers applies to both these log levels.

Reply:
changed to flag on `HttpLogOptions`
 
```java
public HttpLogOptions disableRedactedHeaderLogging(boolean disableRedactedHeaderLogging) {
...
public boolean isRedactedHeaderLoggingDisabled()
```


the motivation was to reduce the volume and `BODY_AND_HEADERS` looks like one where volume does not matter, but I agree that an extra flag is a better long term plan.

I was going to add a flag for query params, but looking into how we log queries (`http://host:port/path/?foo=bar&baz=REDACTED` vs `http://host:port/path/?foo=bar`)  have some doubts if it's useful enough. We can always add it later.